src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -1782,20 +1782,10 @@ public ErrorDiagnostic NestedRuntimePropertyAccessNotSupported(string? resourceS
                 "BCP311",
                 $@"The provided index value of ""{indexSought}"" is not valid for type ""{typeName}"". Indexes for this type must be between 0 and {tupleLength - 1}");
 
-            public ErrorDiagnostic CyclicArmTypeRefs(IEnumerable<string> cycleLinks) => new(
+            public ErrorDiagnostic UnresolvableArmJsonType(string errorSource, string message) => new(
                 TextSpan,
                 "BCP312",
```

Comment:
We've so far tried to avoid re-using error codes, in case it makes errors more difficult to google - please could you pick a new one?

Reply:
The error code being replaced hasn't been included in any releases yet, but that makes a few assumptions about when this PR will be merged and whether we'll need to do a release before then. Updated to use `BCP318` instead.

Reply:
Oh my bad, I missed that. Thanks for addressing anyway!